src/instructlab/process/process.py

```diff
@@ -0,0 +1,384 @@
+    pid = process_registry.processes[local_uuid]["pid"]
+    children_pids = process_registry.processes[local_uuid]["children_pids"]
+    all_processes = [pid] + children_pids
+    for process in all_processes:
+        try:
+            os.kill(process, signal.SIGKILL)
+            logger.info(f"Process {process} terminated.")
+        except (ProcessLookupError, PermissionError):
+            logger.warning(
+                f"Process {process} was not running or could not be stopped."
+            )
+    if remove:
+        process_registry.processes.pop(local_uuid, None)
+    else:
+        process_registry.processes[local_uuid]["done"] = True
+    save_registry(process_registry=process_registry)
+
+
+def list_processes():
+    process_registry = load_registry()
+    if not process_registry:
+        logger.info("No processes currently in the registry.")
+        return
+
+    list_of_processes = []
+    for local_uuid, entry in process_registry.processes.items():
+        all_pids = [entry["pid"]] + entry["children_pids"]
+        if not all_processes_running(all_pids):
+            remove = entry["done"]
+            stop_process(local_uuid=local_uuid, remove=remove)
+            if remove:
+                continue
+        now = datetime.now()
+
+        # Calculate runtime
+        runtime = now - datetime.fromisoformat(entry.get("start_time"))
+        # Convert timedelta to a human-readable string (HH:MM:SS)
+        total_seconds = int(runtime.total_seconds())
+        hours, remainder = divmod(total_seconds, 3600)
+        minutes, seconds = divmod(remainder, 60)
+
+        runtime_str = f"{hours:02}:{minutes:02}:{seconds:02}"
+        list_of_processes.append(
+            (
+                entry.get("type"),
+                entry.get("pid"),
+                local_uuid,
+                entry.get("log_file"),
+                runtime_str,
+            )
+        )
+
+    return list_of_processes
+
+
+def attach_process(local_uuid: str):
+    """
+    Attach to a running process and display its output in real-time.
+
+    Args:
+        local_uuid (str): UUID of the process to attach to
+    """
+    process_registry = load_registry()
+    if local_uuid not in process_registry.processes:
+        logger.warning("Process not found.")
+        return
+
+    process_info = process_registry.processes[local_uuid]
+    log_file = process_info["log_file"]
+
+    if not os.path.exists(log_file):
+        logger.warning(
+            "Log file not found. The process may not have started logging yet."
+        )
+        return
+
+    logger.info(f"Attaching to process {local_uuid}. Press Ctrl+C to detach and kill.")
+    all_pids = [process_info["pid"]] + process_info["children_pids"]
+    if not all_processes_running(all_pids):
+        return
+    try:
+        with open(log_file, "a+") as log:
+            log.seek(0, os.SEEK_END)  # Move to the end of the log file
+            while all_processes_running(all_pids):
```

Comment:
should we continue to stay attached until all child processes are finished instead of exiting immediately after any of them completes?